### ISSUE TITLE
Wrongly resolving plugins in extends

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#9904]
Here:

https://github.com/eslint/eslint/blob/084351bedb1001fc460e9acea4952b0b6139ec23/lib/config/config-file.js#L466

This is the code to resolve the plugin that `extends: "plugin:foo/recommended"` specified. That code looks to use the same logic as shareable config's.

However, the logic to resolve plugins is different to shareable config's. It's:

https://github.com/eslint/eslint/blob/084351bedb1001fc460e9acea4952b0b6139ec23/lib/config/plugins.js#L106

We should fix the resolving logic of plugins in `extends`.

----

Should we address this change as a breaking change?

**Comment by platinumazure** [comment]
Adding "breaking" as a precaution (we can always remove it later if we're sure it's not a breaking change).

**Comment by not-an-aardvark** [comment]
I think this is a similar issue to https://github.com/eslint/eslint/issues/6237, which is blocked by https://github.com/eslint/eslint/issues/3458 because multiple plugins can't have the same name.

**Comment by haraldrudell** [comment]
I think I have this problem

If **eslint** is loaded from an **out-of-project symlinked location**, and an **in-project eslint.rc** file uses extend, the plugin is not found. In particular **eslint-plugin-react**

> 
> plugins:
> \- react

**Comment by not-an-aardvark** [comment]
@haraldrudell I don't think your issue is related to this one. May you please open a new issue?

**Comment by mysticatea** [comment]
I will work on this.
